public/app/plugins/datasource/prometheus/query_hints.ts

```diff
@@ -125,6 +131,17 @@ export function getQueryHints(query: string, series?: any[], datasource?: Promet
     }
   }
 
+  //Check if interval is below safeInterval
+  if (currentInterval && safeInterval && currentInterval === safeInterval) {
```

Comment:
unfortunately this is not precise enough. you can have a query where the `currentInterval` is naturally equal to `safeInterval`, and in such case we do not want to show the warning.

i think we should modify `adjustInterval` so that it by itself returns both the adjusted interval, and a boolean that says if the result was affected by safe-interval or not.